src/device-state/current-state.ts

```diff
@@ -256,7 +255,38 @@ function reportCurrentState(attempts = 0) {
 	})();
 }
 
-export const startReporting = () => {
+export async function startReporting(): Promise<void> {
+	await config.initialized;
+
+	let reportInterval: number;
+	let reportLoop: NodeJS.Timeout;
+
+	// Listen for config changes on stateReportInterval so we can update our intervals
+	config.on('change', (changedConfig) => {
+		if (changedConfig.stateReportInterval) {
+			// Update local copy of reporting interval
+			reportInterval = changedConfig.stateReportInterval;
+			// Update throttled report interval
+			throttledReport = throttle(report, reportInterval);
+			// Stop current report interval
+			clearInterval(reportLoop);
+			// Start interval with new time
+			reportLoop = setInterval(doReport, reportInterval);
+		}
+	});
+
+	try {
+		reportInterval = await config.get('stateReportInterval');
+	} catch {
```

Comment:
I would not catch the `config.get`. I would expect if something fails during `config.get` it should return the default instead of throwing an exception.

Reply:
I don't like the try/catch also, I just copied how Target state fetch is done to work towards both mechanisms behaving the same.